contrib/network/connection_balance/dlb/source/connection_balancer_impl.cc

```diff
@@ -20,9 +20,23 @@ namespace Dlb {
 Envoy::Network::ConnectionBalancerSharedPtr
 DlbConnectionBalanceFactory::createConnectionBalancerFromProto(
     const Protobuf::Message& config, Server::Configuration::FactoryContext& context) {
-  const auto dlb_config = MessageUtil::downcastAndValidate<
-      const envoy::extensions::network::connection_balance::dlb::v3alpha::Dlb&>(
-      config, context.messageValidationVisitor());
+  const auto& typed_config =
+      dynamic_cast<const envoy::config::core::v3::TypedExtensionConfig&>(config);
+  envoy::extensions::network::connection_balance::dlb::v3alpha::Dlb dlb_config;
+  auto status = Envoy::MessageUtil::unpackToNoThrow(typed_config.typed_config(), dlb_config);
+  if (absl::IsInternal(status)) {
+    ExceptionUtil::throwEnvoyException(
+        fmt::format("unexpected dlb config: {}", typed_config.DebugString()));
+  }
+
+  const int num = context.options().concurrency();
+
+  if (num > 32) {
+    ExceptionUtil::throwEnvoyException(
+        "Dlb connection balanncer only supports up to 32 worker threads, "
+        "please decrease the number of threads by `--concurrency`");
```

Comment:
I suggest adding tests for both exceptions.

Reply:
DLB mock functions are still in development, so I want to wait for that and then add all exceptions in tests, are you ok with that?

Reply:
sure.